fdbserver/VersionedBTree.actor.cpp

```diff
@@ -7702,11 +7702,16 @@ class KeyValueStoreRedwood : public IKeyValueStore {
 		        : SERVER_KNOBS->REDWOOD_REMAP_CLEANUP_WINDOW_BYTES;
 
 		EncodingType encodingType = EncodingType::XXHash64;
-
-		// Deterministically enable encryption based on uid
-		if (g_network->isSimulated() && logID.hash() % 2 == 0) {
-			encodingType = EncodingType::XOREncryption;
-			m_keyProvider = std::make_shared<XOREncryptionKeyProvider>(filename);
+		// TODO(yiwu): We should enable encryption only when tenant is in required mode, but currently when existing
+		// storage engine instance is reopen on worker server start, the tenant mode is unknown.
+		if (SERVER_KNOBS->ENABLE_STORAGE_SERVER_ENCRYPTION) {
+			encodingType = EncodingType::AESEncryptionV1;
```

Comment:
Any reason we can't reuse `EncryptUtil::EncryptCipherMode`; EncryptionHeader add this mode details to persistent disk, might be better to keep the usage uniform  (future addition to encryption mode would be easier task, hopefully)
Thoughts?

Reply:
This is Redwood encoding type, where `AESEncryptionV1` is the only available (real) encryption type. If we later support other encryption methods with `BlobCipherEncryptHeader`, maybe we could change to use a more generic naming for the Redwood encryption encoding type.

Reply:
Yeah, I was trying to see if we can avoid having a separate encoding type for Redwood and just leverage the one supported by the BlobCipher, as types defined there are supported for all FDB processes...

Reply:
The redwood encoding type is not necessary about encryption, but about how a redwood page get encoding. For example the default ending type XXHash64 means the redwood pages are with xxhash64 checksum, and non-encrypted.